interpreter/interpreter.go

```diff
@@ -496,6 +496,11 @@ func (itrp *Interpreter) doStringPart(ctx context.Context, part semantic.StringE
 			return nil, errors.Newf(codes.Invalid, "%s: interpolated expression produced a null value",
 				p.Location())
 		} else if v.Type().Nature() != semantic.String {
+			val := values.NewString(fmt.Sprintf("%v", values.Unwrap(v)))
```

Comment:
Do we want to rely on Go's formatting of the value here? Or should we be more explicit about how each of the Stringable types is converted to a string? For example using `strconv.FormatInt` etc?

Reply:
I think either can work. As far as I could tell when testing, the actual output didn't differ for the basic types. I can change it just in case that behavior changes in the future.

Reply:
I think we should be explicit about how we are formatting the values. I don't expect Go's definition to change, but it would be best to be explicit so we are not tied to Go's behaviors.